ethereum/migrations/2_next.js

```diff
@@ -63,5 +64,30 @@ module.exports = function(deployer, network, accounts) {
     );
 
     await deployer.deploy(TestToken, 100000000, "Test Token", "TEST");
+
+    // Deploy ERC1820 Registry for our E2E stack.
```

Comment:
whats this for? seems unused?

Reply:
The ERC1820 registry contract is a requirement for ERC777 tokens. Its fundamental to how ERC777 can implement token transfers & notification within a single transaction.

It seems unused, but actually the ERC777 contract references the deployed contract using a hardcoded address:
https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC777/ERC777.sol#L31

Reply:
Also to add, The registry is already deployed to mainnet and ropsten. So that's why we only need to deploy it for E2E tests.